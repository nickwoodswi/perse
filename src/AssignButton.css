.assignment-preview{
    width: 100%;
    max-width: 505px;
    background-color: rgba(0,0,0, 0.9);
    text-align: center;
    padding-bottom: 15px;
    padding-top: 15px;
    margin-bottom: 15px;
    color: white;
}
    ul {
        list-style: none;
        padding-left: 0;
        margin: 0;
    }
.assign-workout-button{
    background-color: white;
    margin-left: 0;
    border-radius: 5px;
    border: none;
    box-shadow: 2px 2px 2px darkgrey;
    font-size: 16px;
}
    .assign-workout-button:active{
        background-color: greenyellow
    }