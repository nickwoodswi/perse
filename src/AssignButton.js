import React, { Component } from 'react'
import Recurrance from './Recurrance'
import './AssignButton.css'

class AssignButton extends Component {
    render() {
        if (!this.props.dayEnd ||
            !this.props.dayStart || 
            !this.props.monthEnd ||
            !this.props.monthStart ||
            !this.props.firstName ||
            !this.props.lastName ||
            !this.props.workoutName ||
            !this.props.yearEnd ||
            !this.props.yearStart ||
            !this.props.recurrance) {
                return(<></>)
        } else {

        return(
            <>
            <div className="assignment-preview">
                <ul>
                    <li>Assigning <b>"{this.props.workoutName}"</b></li>
                    <li>to <b><i>{this.props.firstName} {this.props.lastName}</i></b></li>
                    <Recurrance recurrance={this.props.recurrance} />
                    <li>from <b><i>{this.props.monthStart} {this.props.dayStart}, {this.props.yearStart}</i></b></li>
                    <li>thru <b><i>{this.props.monthEnd} {this.props.dayEnd}, {this.props.yearEnd}</i></b></li>
                </ul>
            </div>
            <button className="assign-workout-button" onClick={this.props.assignWorkout}>
                ASSIGN WORKOUT
            </button>
            </>
        )
        
        }
    }
}

export default AssignButton