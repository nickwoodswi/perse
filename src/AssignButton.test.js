import React from 'react';
import { render } from '@testing-library/react';
import AssignButton from './AssignButton';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders AssignButton', () => {
    render(<MemoryRouter><AssignButton /></MemoryRouter>);
});

it('AssignButton renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><AssignButton /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});