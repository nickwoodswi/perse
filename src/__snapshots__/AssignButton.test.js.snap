// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignButton renders the UI as expected 1`] = `null`;
